### PR TITLE
Move classes from org.gradle.util to org.gradle.util.internal

### PR DESCRIPTION
**Pull request by donat**
This PR moves the following classes from `org.gradle.util` to the `org.gradle.util.internal` package:
- `VersionNumber`
- `TextUtil`
- `BuildCommencedTimeProvider`
- `TreeVisitor`
- `WrapUtil`
- `Mark SingleMessageLogger for removal`
- `IncubationLogger`
- `RelativePathUtil`
- `AntUtil`
- `DistributionLocator`
- `JarUtil`
- `NoOpChangeListener`
- `DeferredUtil`

The PR ensure backwards compatibility for all classes that have usages ([example](https://grep.app/search?q=org.gradle.util.ConfigureUtil)) in GitHub (VersionNumber, TextUtil, WrapUtil, RelativePathUtil, DistributionLocator): 
- the original class is kept in the `org.gradle.util` package
- all internal usages are removed
- the class is marked as `@deprecated` and set for removal in Gradle 8.0

The PR doesn't tackle all classes in `org.gradle.util`, The remaining items will be handled in a follow-up PR or PRs.

Note to reviewer: The raw changeset huge. It's easier to do the review if you check the changes commit-by-commit.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test quickFeedbackLinux

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@rieske I've resolved all merge conflicts; this PR is good to go on my side.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -110,11 +119,27 @@ The following classes were moved from `org.gradle.util` to the `org.gradle.util.
...
-Also, the unused classes were deleted from `org.gradle.util`:
+The last set of classes have no external or internal usages and therefore were deleted:
 
 - `DiffUtil`
 - `NoopChangeListener`
 - `EnumWithClassBody`
 - `AlwaysTrue`
 - `ReflectionEqualsMatcher`
 - `DynamicDelegate`
+- `TreeVisitor`
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:130]: I think this one was moved to internal package and not deleted.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 2]
```diff
@@ -110,11 +119,27 @@ The following classes were moved from `org.gradle.util` to the `org.gradle.util.
...
+- `SingleMessageLogger`
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:131]: This one is marked deprecated, not deleted yet.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 3]
```diff
@@ -110,11 +119,27 @@ The following classes were moved from `org.gradle.util` to the `org.gradle.util.
...
+- `IncubationLogger`
+- `AntUtil`
+- `JarUtil`
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:134]: Those two were moved to internal.

**Comment by donat** [comment]
I had to revert `Path` because it breaks the test retry plugin. I'll handle that in a separate PR.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 1 above)
Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Fixed

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 2 above)
Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Fixed

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 3 above)
Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Fixed. Thanks for verifying those!

**Comment by donat** [comment]
@bot-gradle test rFM

**Review by rieske — APPROVED** [review]

**Comment by donat** [comment]
@bot-gradle test rFM

**Pull request merged**